packages/sitecore-jss/src/utils/utils.ts

```diff
@@ -161,17 +161,109 @@ export const enforceCors = (
   return false;
 };
 
-  /**
-   * Generates all possible permutations of an array of key-value pairs.
-   * This is used to create every possible combination of URL query parameters.
-   * @param {Array<[string, string]>} array - The array of key-value pairs to permute.
-   * @returns {Array<Array<[string, string]>>} - A 2D array where each inner array is a unique permutation of the input.
-   */
- export const getPermutations = (array: [string, string][]): [string, string][][] =>{
-  if (array.length <= 1) return [array];
-
-  return array.flatMap((current, i) => {
-    const remaining = array.filter((_, idx) => idx !== i);
-    return getPermutations(remaining).map((permutation) => [current, ...permutation]);
-  });
+/**
+ * Determines whether the given input is a regular expression or resembles a URL.
+ * @param {string} input - The input string to evaluate.
+ * @returns {'regex' | 'url'} - Returns 'url' if the input looks like a URL, otherwise 'regex'.
+ */
+export const isRegexOrUrl = (input: string): 'regex' | 'url' => {
+  // Remove the trailing slash.
+  input = input.slice(0, -1);
+
+  // Check if the string resembles a URL.
+  const isUrlLike = /^\/[a-zA-Z0-9\-\/]+(\?([a-zA-Z0-9\-_]+=[a-zA-Z0-9\-_]+)(&[a-zA-Z0-9\-_]+=[a-zA-Z0-9\-_]+)*)?$/.test(input);
```

Comment:
Is it possible to identify using `new URL(str)`? It might throw an error if string is not a url

Reply:
That's not coverage all tests for example `/path/.*`(it's regular expression), if we use `new URL` it notify us that it's url instead of regex